src/contracts/libraries/Checkpoints.sol

```diff
@@ -0,0 +1,287 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin-upgrades-v4.9.0/contracts/utils/math/MathUpgradeable.sol";
+import "@openzeppelin-upgrades-v4.9.0/contracts/utils/math/SafeCastUpgradeable.sol";
+
+/**
+ * @title Library for handling checkpoints as part of allocating and slashing.
+ * @notice This library is using OpenZeppelin's CheckpointsUpgradeable library (v4.9.0)
+ * and removes structs and functions that are unessential.
+ * Interfaces and structs are renamed for clarity and usage (timestamps, etc).
+ * Some additional functions have also been added for convenience.
+ * @dev This library defines the `History` struct, for checkpointing values as they change at different points in
+ * time, and later looking up past values by block number. See {Votes} as an example.
+ *
+ * To create a history of checkpoints define a variable type `Checkpoints.History` in your contract, and store a new
+ * checkpoint for the current transaction block using the {push} function.
+ *
+ * _Available since v4.5._
+ */
+library Checkpoints {
```

Comment:
mentioned in PR already, but IMO we should not have a library/file called `Checkpoints.sol`. Sorry, PEPE is squatting that name now

`Snapshots` would be perfect though!

Reply:
You win, lets go with `Snapshots`